File: TwoSum_SortedArray/twosum.dart
class Solution {
  List<int> twoSum(List<int> numbers, int target) {
    int i = 0;
    int j = numbers.length - 1;

    while (i < j) {
      int sum = numbers[i] + numbers[j];

      if (sum == target) {
        return [i + 1, j + 1];
      } else if (sum > target) {
        j--;
      } else {
        i++;
      }
    }

    return [-1, -1];
  }
}

File: TwoSum_SortedArray/twosum.go
package twosumsortedarray

func twoSum(numbers []int, target int) []int {
	if len(numbers) == 2 {
		return []int{1, 2}
	}

	i, j := 0, len(numbers)-1

	for i < j {
		sum := numbers[i] + numbers[j]

		if sum == target {
			break
		} else if sum > target {
			j--
		} else {
			i++
		}
	}

	return []int{i + 1, j + 1}
}
